lib/lua/finishJob.lua

```diff
@@ -0,0 +1,47 @@
+
+--[[
+key 1 -> bq:name:active
+key 2 -> bq:name:waiting
+key 3 -> bq:name:stalling
+key 4 -> bq:name:failed
+key 5 -> bq:name:delayed
+key 6 -> bq:name:earlierDelayed
+key 7 -> bq:name:succeeded
+key 8 -> bq:name:events
+key 9 -> bq:name:jobs
+
+arg 1 -> job id
+arg 2 -> remove job?
+arg 3 -> job data
+arg 4 -> new job status
+arg 5 -> delayed time
+arg 6 -> job event data
+--]]
+
+-- If we don't find the job in the active list, ensure it hasn't been recently stalled
+if not redis.call("lrem", KEYS[1], 0, ARGV[1]) then
```

Comment:
according to [the docs](https://www.lua.org/pil/3.3.html), lua considers `0` to be truthy - I think we need to use a comparison operator instead?

Reply:
Ok, better make sure I add a test that proves this.

Reply:
@skeggse -- sorry it took me so long to get back to this. Any advice on how to achieve this? 

I tried the following but it doesn't seem to move the job into the `waiting` list (i.e the final assertion fails):

```js
      const queue = t.context.makeQueue({
        stallInterval: 1,
      });

      const job = queue.createJob({foo: 'bar'});
      await job.save();

      // Block the job from finishing
      const {done, next} = reef(1);
      queue.process(async () => {
        await done;
      });

      // Mark the jobs as stalling, so that Queue#process immediately detects
      // them as stalled.
      await forceStall(queue);
      await queue._doStalledJobCheck();

      const counts = await queue.checkHealth();
      t.is(counts.waiting, 1);
```